woocommerce/google-listings-and-ads
# Move the pre-launch checklist from step 3 to 4 for the MC setup

### Changes proposed in this Pull Request:

Partial of #863 and based on #926

- Move `js/src/setup-mc/setup-stepper/setup-free-listings/pre-launch-checklist` in the folder of `js/src/setup-mc/setup-stepper/store-requirements/`
- Separate the pre-launch checks into another validation function
- Move the pre-launch checklist from step 3 to 4 for the MC setup and relevant checking and data updating

### Screenshots:

https://user-images.githubusercontent.com/17420811/127865542-96a7a46b-c524-4cd7-be06-a9e25f949e00.mp4

### Detailed test instructions:

1. Go to the MC setup and proceed to step 3
2. The pre-launch checklist should no longer be in step 3
3. Change some product listings settings to see if the disable state of "Continue" button works correctly
4. Proceed to step 4 and the pre-launch checklist should be at the bottom
5. Fill in the contact information
6. Tick or untick each pre-launch checklist to see if the disable state of "Complete setup" button works correctly

### Changelog entry
